activerecord/lib/active_record/fixtures.rb

```diff
@@ -1028,9 +1055,21 @@ def enlist_fixture_connections
     end
 
     private
-      def load_fixtures(config)
-        fixtures = ActiveRecord::FixtureSet.create_fixtures(fixture_path, fixture_table_names, fixture_class_names, config)
-        Hash[fixtures.map { |f| [f.name, f] }]
+      def load_fixtures(config, environment)
+        original_db_config = config.connection_config
+        all_fixtures = config.configurations.configs_for(env_name: environment).map do |db_config|
+          config.establish_connection(db_config.config)
+          fixtures = ActiveRecord::FixtureSet.create_fixtures(
+            config.connection.fixtures_path,
+            fixture_table_names[db_config.spec_name],
+            fixture_class_names,
+            config,
+          )
```

Comment:
This is starting to get complex, wondering if it makes sense to pull fixtures into a separate method. Thoughts?

Reply:
Agree, I'll try to break up the complexity a little more. I'm a little hesitant to add too many private methods to a concern though.